airflow/www/utils.py

```diff
@@ -374,6 +376,30 @@ def get_chart_height(dag):
     return 600 + len(dag.tasks) * 10
 
 
+def get_python_source(x: Union[Callable, str]) -> str:
+    """
+    Helper function to get Python source (or not), preventing exceptions
+    """
+    if isinstance(x, str):
+        return x
+    source_code = None
+    if isinstance(x, functools.partial):
+        source_code = inspect.getsource(x.func)
+    if source_code is None:
+        try:
+            source_code = inspect.getsource(x)
+        except TypeError:
+            pass
+    if source_code is None:
+        try:
+            source_code = inspect.getsource(x.__call__)
```

Comment:
Why do we need this?

Reply:
This fn was copied from Airflow 1.10.4. It was removed from master.
https://github.com/apache/airflow/blob/1.10.4/airflow/www/utils.py#L407

I did a test to confirm it is for a class that implements __call__:
import inspect

class A:
  def __call__(self, x):
    return x+1

a = A()
print(inspect.getsource(a.__call__)) # it works
print(inspect.getsource(a)) # it does not